os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -181,6 +181,8 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 	loadinfo->dataalloc = loadinfo->textalloc + loadinfo->textsize;
 #endif
 	loadinfo->binp->heapstart = loadinfo->dataalloc + datasize;
+	loadinfo->binp->uheap = (struct mm_heap_s *)loadinfo->binp->heapstart;
```

Comment:
How about directly setting the value of uheap and remove the heapstart variable?

`loadinfo->binp->uheap = (struct mm_heap_s *)(loadinfo->dataalloc + datasize);`